samples/corda/corda-simple-application/workflows-kotlin/src/main/kotlin/com/example/flow/AssetFlow.kt

```diff
@@ -624,3 +632,204 @@ class AcceptorOfTransfer(val otherPartySession: FlowSession) : FlowLogic<SignedT
     }
 }
 
+/**
+ * The GetAssetJsonStringFromStatePointer flow fetches the asset state from its state pointer [AssetPledgeState].assetStatePointer
+ * and creates marshalled JSON encoded object which is returned.
+ * This function is called by the exporting network in the context of interop-query from the importing network
+ * to the exporting network before performing claim on remote asset.
+ *
+ * @property assetPledgeState The (interop) vault state that represents the pledge details on the ledger.
+ */
+@InitiatingFlow
+@StartableByRPC
+class GetAssetJsonStringFromStatePointer(
+    val assetPledgeState: AssetPledgeState
+) : FlowLogic<String>() {
+    @Suspendable
+    override fun call(): String {
+
+        if (assetPledgeState.assetStatePointer == null) {
+            // Typically, [AssetPledgeState].assetStatePointer will be null only in the case of pledge details not
+            // being available for a given pledgeId. The flow GetAssetPledgeStatus in AssetTransferFlows sets this
+            // pointer to null if the pledge-state is not available in the context of the interop-query from the
+            // importing n/w to the exporting n/w. Hence return empty string, and this will not be passed to the
+            // JSON unmarshalling method GetTokenStateAndContractId since the expiryTime will be elapsed for the
+            // claim to happen (i.e., if assetStatePointer is null, then expiryTimeSecs will be set to past time).
+            return ""
+        }
+
+        val assetStatePointer: StaticPointer<ContractState> = assetPledgeState.assetStatePointer!!
+        val assetState = assetStatePointer.resolve(serviceHub).state.data as AssetState
+
+        println("Creating simple asset JSON from StatePointer.")
+        var marshalledAssetJson =
+            subFlow(MarshalFungibleAsset(assetState.tokenType, assetState.quantity, assetPledgeState.lockerCert))
+
+        return marshalledAssetJson
+    }
+}
+
+/**
+ * The GetAssetPledgeStatusByPledgeId flow fetches the fungible asset pledge status in the exporting network and returns as byte array.
+ * It is called during the interop query by importing network before performing the claim on fungible asset pledged in exporting network.
+ *
+ * @property pledgeId The unique identifier representing the pledge on an asset for transfer, in the exporting n/w.
+ * @property recipientNetworkId The id of the network in which the pledged asset will be claimed.
+ */
+@InitiatingFlow
+@StartableByRPC
+class GetAssetPledgeStatusByPledgeId(
+    val pledgeId: String,
+    val recipientNetworkId: String
+) : FlowLogic<ByteArray>() {
+    @Suspendable
+    override fun call(): ByteArray {
+
+        var assetPledgeState: AssetPledgeState = subFlow(GetAssetPledgeStatus(pledgeId, recipientNetworkId))
+        println("Obtained [AssetPledgeState] vault state: ${assetPledgeState}.\n")
+        val marshalledAssetJson = subFlow(GetAssetJsonStringFromStatePointer(assetPledgeState))
```

Comment:
This subFlow also seems is called only once, can the contents of the subFlow moved inside this flow directly?

This can't be made an independent function, so the only option is to move the contents inside this flowLogic.